api/src/repositories/project-repository.ts

```diff
@@ -28,185 +29,146 @@ const defaultLog = getLogger('repositories/project-repository');
-        AND p.project_id IN (
-          SELECT
-            project_id
-          FROM
-            project_participation
-          where
-            system_user_id = ${systemUserId}
-        )
-      `);
+      query.whereIn('p.project_id', (subQueryBuilder) => {
+        subQueryBuilder.select('project_id').from('project_participation').where('system_user_id', systemUserId);
+      });
     }
 
-    if (filterFields && Object.keys(filterFields).length !== 0 && filterFields.constructor === Object) {
-      if (filterFields.start_date && !filterFields.end_date) {
-        sqlStatement.append(SQL` AND p.start_date >= ${filterFields.start_date}`);
-      }
-
-      if (!filterFields.start_date && filterFields.end_date) {
-        sqlStatement.append(SQL` AND p.end_date <= ${filterFields.end_date}`);
-      }
+    // Start Date filter
+    if (filterFields.start_date) {
+      query.andWhere('p.start_date', '>=', filterFields.start_date);
+    }
 
-      if (filterFields.start_date && filterFields.end_date) {
-        sqlStatement.append(
-          SQL` AND p.start_date >= ${filterFields.start_date} AND p.end_date <= ${filterFields.end_date}`
-        );
-      }
+    // End Date filter
+    if (filterFields.end_date) {
+      query.andWhere('p.end_date', '<=', filterFields.end_date);
+    }
 
-      if (filterFields.project_name) {
-        sqlStatement.append(SQL` AND p.name = ${filterFields.project_name}`);
-      }
+    // Project Name filter (exact match)
+    if (filterFields.project_name) {
+      query.andWhere('p.name', filterFields.project_name);
+    }
 
-      if (filterFields?.species && filterFields?.species?.length > 0) {
-        sqlStatement.append(SQL` AND sp.wldtaxonomic_units_id =${filterFields.species[0]}`);
-      }
+    // Focal Species filter
+    if (filterFields.species_tsns?.length) {
+      query.whereIn('sp.itis_tsn', filterFields.species_tsns);
+    }
 
-      if (filterFields.keyword) {
-        const keyword_string = '%'.concat(filterFields.keyword).concat('%');
-        sqlStatement.append(SQL` AND p.name ilike ${keyword_string}`);
-        sqlStatement.append(SQL` AND fs.name ilike ${keyword_string}`);
-        sqlStatement.append(SQL` OR s.name ilike ${keyword_string}`);
-      }
+    // Keyword Search filter
+    if (filterFields.keyword) {
+      const keywordMatch = `%${filterFields.keyword}%`;
+      query.where((subQueryBuilder) => {
+        subQueryBuilder
+          .where('p.name', 'ilike', keywordMatch)
+          .orWhere('p.objectives', 'ilike', keywordMatch)
+          .orWhere('s.name', 'ilike', keywordMatch);
+      });
     }
 
-    sqlStatement.append(SQL`
-      group by
-        p.project_id,
-        p.name,
-        p.start_date,
-        p.end_date
-    `);
+    // Programs filter
+    if (filterFields.project_programs?.length) {
+      query.havingRaw(
+        `array_agg(DISTINCT prog.program_id) && ARRAY[${filterFields.project_programs.join(',')}]::integer[]`
+      );
```

Comment:
I think this is more ideal, since `having` is less performant. I tried it locally and it seems to be working, maybe just double check in case:

```suggestion
      query.where('prog.program_id', 'IN', filterFields.project_programs);
```

Reply:
(I've noticed in past as well, that knex `whereIn` doesn't seem to work due to some typing issues. But if you just use `where` + `IN` then it works fine, and still recognizes it has requiring an array for the third param, etc)

Reply:
> (I've noticed in past as well, that knex `whereIn` doesn't seem to work due to some typing issues. But if you just use `where` + `IN` then it works fine, and still recognizes it has requiring an array for the third param)

ahh that's interesting. I tried `.whereIn` and it didn't work (ostensibly for the same reason you mentioned). I will change it